### PR TITLE
8065422: Trailing dot in hostname causes TLS handshake to fail with SNI disabled

### PR DESCRIPTION
**Pull request by XueleiFan**
A hostname in an URL ending with a dot is valid (See RFC 1034).  However, it is not a valid SNI hostname.  The ending dot should be ignored while checking the hostname with SNI or the name in a X.509 certificate.

The update could be verified with jshell.
$ $JDK_HOME/bin/jshell
jshell> URL url = new URL("https://www.google.com./");
jshell> URLConnection conn = url.openConnection();
jshell> conn.connect();

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8065422](https://bugs.openjdk.java.net/browse/JDK-8065422): Trailing dot in hostname causes TLS handshake to fail with SNI disabled

### Reviewers
 * [Weijun Wang](https://openjdk.java.net/census#weijun) (@wangweij - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/7205/head:pull/7205` \
`$ git checkout pull/7205`

Update a local copy of the PR: \
`$ git checkout pull/7205` \
`$ git pull https://git.openjdk.java.net/jdk pull/7205/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 7205`

View PR using the GUI difftool: \
`$ git pr show -t 7205`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/7205.diff">https://git.openjdk.java.net/jdk/pull/7205.diff</a>

</details>

### CODE REVIEW

**Comment by wangweij** [comment]
Is it possible to add a regression test using the `-Djdk.net.hosts.file` feature? It is a JVM-only `/etc/hosts` alternative.

**Comment by XueleiFan** [comment]
> Is it possible to add a regression test using the `-Djdk.net.hosts.file` feature? It is a JVM-only `/etc/hosts` alternative.
Good to know that.  I added the test with customized hosts, and use URL "www.example.com.".

**Comment by XueleiFan** [comment]
ping ...

File: test/jdk/javax/net/ssl/ServerName/EndingDotHostname.java [hunk 1]
```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright (C) 2022 THL A29 Limited, a Tencent company. All rights reserved.
+ * DO NOT ALTER OR REMOVE NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation. THL A29 Limited designates
+ * this particular file as subject to the "Classpath" exception as provided
```
Comment (wangweij) [inline test/jdk/javax/net/ssl/ServerName/EndingDotHostname.java:8]: Please remove the Classpath part. It does not apply to tests. Same comment to the other file.

Comment (XueleiFan) [reply test/jdk/javax/net/ssl/ServerName/EndingDotHostname.java]: Ooops, I missed it.  Thank you!  Updated.

**Review by wangweij — APPROVED** [review]

**Pull request closed without merging**